lib/passport/strategy.js

```diff
@@ -1,36 +1,25 @@
 var
-  q = require('q'),
   LocalStrategy = require('passport-local').Strategy;
 
 module.exports = function(context) {
   this.verify = function(username, password, done) {
-    var
-      deferred = q.defer(),
-      repositories = context.accessors.repositories;
-
-    repositories.user.load(username)
+    context.accessors.users.load(username)
       .then(userObject => {
         if (userObject !== null) {
           context.passwordManager.checkPassword(password, userObject.password || userObject._source.password)
             .then(result => {
               if (result === false) {
-                deferred.reject(new context.errors.ForbiddenError('Bad Credentials'));
-              }
-              else {
-                deferred.resolve(userObject);
+                return done(new context.errors.ForbiddenError('Bad Credentials'));
               }
+
+              done(null, userObject);
             });
         }
         else {
-          deferred.reject(new context.errors.ForbiddenError('Bad Credentials'));
+          done(new context.errors.ForbiddenError('Bad Credentials'));
```

Comment:
didn't really paid attention to the existing code, but I agree with you, so I changed the message to "login failed" :)